src/main/utils/shell-env/compute-shell-environment.injectable.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import type { AsyncResult } from "../../../common/utils/async-result";
+import { getInjectable } from "@ogre-tools/injectable";
+import isWindowsInjectable from "../../../common/vars/is-windows.injectable";
+import { disposer } from "../../../common/utils";
+import computeUnixShellEnvironmentInjectable from "./compute-unix-shell-environment.injectable";
+
+export type EnvironmentVariables = Partial<Record<string, string>>;
+export type ComputeShellEnvironment = (shell: string) => Promise<AsyncResult<EnvironmentVariables | undefined, string>>;
+
+const computeShellEnvironmentInjectable = getInjectable({
+  id: "compute-shell-environment",
+  instantiate: (di): ComputeShellEnvironment => {
+    const isWindows = di.inject(isWindowsInjectable);
+    const computeUnixShellEnvironment = di.inject(computeUnixShellEnvironmentInjectable);
+
+    if (isWindows) {
+      return async () => ({
+        callWasSuccessful: true,
+        response: undefined,
+      });
+    }
```

Comment:
This is the main part of the code. In #6351 it was changed to return `{}` instead of `process.env` which breaks the usage of this injectable within `ShellSession`.